src/osd/OSD.cc

```diff
@@ -1333,6 +1332,7 @@ OSDMapRef OSDService::try_get_map(epoch_t epoch)
   if (epoch > 0) {
     dout(20) << "get_map " << epoch << " - loading and decoding " << map << dendl;
     bufferlist bl;
+    Mutex::Locker l(map_cache_lock);
```

Comment:
i think the reason why we have a grand `map_cache_lock` is to make sure the consistency of map caches (`map_cache`, `map_bl_cache` and `map_bl_inc_cache`) maintained by OSDServices. as we take them a single cache when trying to   read an osdmap. and as you know, both SharedLRU and SimpleLRU have its own lock.

so we can safely drop this lock if we don't care about performance at all.

but if the performance comes into the picture, we might want to weigh the cost of acquiring the grand lock and the cost of acquiring the locks in caches and possibly cost of unnecessary loading the osdmaps from ObjectStore and decoding them.

so i don't think we can draw a conclusion that this change will be a win in performance just by inspecting the code.

do you have a benchmark to support your change?